enterprise/server/ociregistry/ociregistry.go

```diff
@@ -190,20 +215,178 @@ func (r *registry) handleBlobsOrManifestsRequest(ctx context.Context, w http.Res
 	}
 	defer upresp.Body.Close()
 
-	if upresp.Header.Get(headerWWWAuthenticate) != "" {
-		w.Header().Add(headerWWWAuthenticate, upresp.Header.Get(headerWWWAuthenticate))
+	for _, header := range []string{headerContentLength, headerContentType, headerDockerContentDigest, headerWWWAuthenticate} {
+		if upresp.Header.Get(header) != "" {
+			w.Header().Add(header, upresp.Header.Get(header))
+		}
 	}
-	w.Header().Add(headerContentType, upresp.Header.Get(headerContentType))
-	w.Header().Add(headerDockerContentDigest, upresp.Header.Get(headerDockerContentDigest))
-	w.Header().Add(headerContentLength, upresp.Header.Get(headerContentLength))
 	w.WriteHeader(upresp.StatusCode)
-	_, err = io.Copy(w, upresp.Body)
+
+	contentLength, err := strconv.ParseInt(upresp.Header.Get(headerContentLength), 10, 64)
+	hasContentLength := err == nil
```

Comment:
Not loving this use of `err == nil` to indicate presence of header because it's ignoring errors. 
Please explicitly check if the header is "" and properly handle the case if the header is present and has an unparsable value. Same for digest below.

Reply:
Good call. Made a change to check for presence of header values and whether they look valid.